### PR TITLE
Bug 1981090: [ON-PREM] HAProxy - enable listening sockets retrieval from old processes

### PR DESCRIPTION
**Pull request by yboaron**
To use the -x option [1] in HAProxy the socket must be enabled using "expose-fd listeners" in haproxy configuration.

[1] https://github.com/haproxy/haproxy/blob/c3fe968f221e2f7c63b7eb24f557223c4bbbe75f/doc/management.txt#L325

### CODE REVIEW

**Review by mandre — COMMENTED** [review]
/lgtm

**Review by kikisdeliveryservice — APPROVED** [review]
Adding approval, remove hold when you have the LGTMs you need @yboaron 

/hold

**Comment by cybertron** [comment]
/lgtm
/hold cancel

We have lgtm and ci pass from all platforms.

**Comment by cybertron** [comment]
Hmm, agnostic-upgrade looks pretty red lately. Is that a known issue?

**Pull request merged**
